core/src/impl/Kokkos_Utilities.hpp

```diff
@@ -411,6 +411,16 @@ struct inclusive_scan_integer_sequence
 
 }} // namespace Kokkos::Impl
 
+namespace Kokkos {
+
+template <class T>
+KOKKOS_INLINE_FUNCTION void swap(T& a, T& b) {
+  T c{std::move(a)};
+  a = std::move(b);
+  b = std::move(c);
```

Comment:
Shouldn't you be using `Kokkos::move()`?

Reply:
`std::move` actually works in device code with recent CUDA versions, not sure if we still support a CUDA version where it doesn't work. I can check on that.

Reply:
How recent is recent?

Reply:
I put `std::move` in `Kokkos::View` a while ago, so basically its required now. Our tests are still passing on CUDA 8 and 9 versions.